### ISSUE TITLE
Formatting new posts

### ISSUE THREAD

**Issue by AlexKucera** [sebsel/kirby-micropub#17]
Hi Sebastian,

I am having a hard time wrapping my head around how to format new posts.

By default the plugin creates:

```
Text: Testing MicroPub posting from the micro.blog iOS app.

----

Client: https://micro.blog/

----

Updated: 2017-07-10 11:35:18

----

Date: 2017-07-10 11:35:18
```

what I want/need is this though:

```
Title: —

----

Text: Testing MicroPub posting from the micro.blog iOS app.

----

Date: 10.07.2017

----

Time: 2017-07-10-11-35-18

----

Client: https://micro.blog/

----

Updated: 2017-07-10 11:35:18

```

So it's mainly the two different data formats and the extra (mandatory) title field, which I use for RSS purposes.

Can you point me in the right direction?

My Kirby config file looks like so:

```
/* micropub */

c::set('micropub.page-creator', function($uid, $template, $data) {
  // Rename fields (you can also rename Kirby's fields)
  if (isset($data['name'])) $data['title'] = "-";
  $data['date'] = $data['published'];

  // No double fields
  unset($data['name'], $data['published']);

  // Add new entry to the blog
  $newEntry = page('microblog')->children()->create($uid, 'microblog', $data);

  // Make it visible
  $newEntry->sort(date('Ymd', strtotime($data['date'])));

  // Return the new entry
  return $newEntry;
});
```

**Comment by sebsel** [comment]
I don't use the panel myself, so I do not need to have a title-field. If you want it to be a `-`, go with this: 
```php
$data['title'] = isset($data['name']) ? $data['name'] : '-';
```
In this case, it uses the Micropub 'name' property as the title if it is available, and use `-` if not. In your code, you're just throwing it away :)

What I don't get, is why you need:
```
Date: 10.07.2017

----

Time: 2017-07-10-11-35-18
```

The better way of storing it is `Date: 2017-07-10 11:35:18`, which is why it gives you that. You can use the [datetime](https://getkirby.com/docs/cheatsheet/panel-fields/datetime) panel field to combine date and time into one field. Due to the way Kirby's (date)time field works, it will throw away the minutes and seconds though, so set `interval` to 60 to at least keep the minute. (I am not a fan of the panel in that regard.)

You can then use `$page->date('d.m.Y');`to get the `10.07.2017` notation in your templates. (And `$page->date('d.m.Y h:i:s', 'updated');` to get the time of the update.) 

You can also remove `$data['published']` from the `unset()`, so you keep that data around.

**Comment by AlexKucera** [comment]
Yeah, I don't have a good explanation for the double date. It's a legacy thing from my pre-Kirby days and when I jumped into Kirby 1.0 I didn't know any better. Also I think Kirby 1 didn't have datatime fields come to think of it. And by now there are so many pages on my blog that I don't feel like cleaning it up. I know. Bad me. :)

Thanks I'll give your suggestions a try.

**Comment by sebsel** [comment]
Ah! In that case I would go for something like this:

```php
c::set('micropub.page-creator', function($uid, $template, $data) {
  // Rename fields (you can also rename Kirby's fields)
  $data['title'] = isset($data['name']) ? $data['name'] : '-';
  $time = strtotime($data['published']);
  $data['date'] = date('d.m.Y', $time);
  $data['time'] = date('Y-m-d-H-i-s', $time);

  // No double fields
  unset($data['name']);

  // Add new entry to the blog
  $newEntry = page('microblog')->children()->create($uid, 'microblog', $data);

  // Make it visible
  $newEntry->sort(date('Ymd', $time));

  // Return the new entry
  return $newEntry;
});
```

That keeps the original date around in `published`, but still uses the `10.07.2017` and `2017-07-10-11-35-18` notations in `date` and `time` :)

**Comment by AlexKucera** [comment]
That did it! Thanks Sebastian.
